content/tutorials/mapping.md

```diff
@@ -0,0 +1,172 @@
+---
+title: "Using SLAM for Map Building"
+date: 2021-03-20T17:15:01-07:00
+summary: Create custom maps with your car
+difficulty: Intermediate
+duration: 0
+featured: true  # whether this is listed at / (must also be top 6 by weight)
+active: true    # whether this is listed at /tutorials/
+draft: false      # whether Hugo considers this a draft
+weight: 100
+---
+
+## By: [Madison Doerr](https://mcdoerr.github.io/) and [Markus Schiffer](www.linkedin.com/in/markusschiffer)
+
+### Introduction
+The MuSHR car uses a map of its environment to localize itself. We can create 
+custom 2D maps for our cars by using [gmapping](http://wiki.ros.org/gmapping) and our car's laser scanner to 
+survey the area to be mapped.
+
+{{< figure src="/tutorials/mapping/map-demo.png" width="500" >}}
+
+### Goal
+Create a custom map of an area by driving the car to survey the surroundings.
+
+### Requirements
+  - Complete the [quickstart](/tutorials/quickstart) tutorial
+  - Complete the [first steps](/tutorials/first_steps) tutorial
+  - A desktop/laptop that can ssh into the car
+
+### Ssh into the car
+Boot up the car and connect the controller. 
+
+Ssh into the car on your computer, using the following command while connected to the ROBOT_AP network.
+```bash
+$ ssh robot@10.42.0.1 -X
+```
+Or, if you set up Wifi connect in the [first steps](/tutorials/first_steps) tutorial, use the following command while connected to the same wifi network as your robot. Replace the IP listed below with the IP of your robot.
+```bash
+$ ssh robot@172.16.77.37 -X
+```
+
+### Update/Install gmapping
+In a terminal sshed into the car using the above steps, make sure gmapping is installed.
+
+If you installed ros-kinetic in the quickstart tutorial:
+```bash
+$ sudo apt install ros-kinetic-gmapping
+```
+If you installed ros-melodic in the quickstart tutorial:
+```bash
+$ sudo apt install ros-melodic-gmapping
+```
+
+### Start teleop and gmapping
+Once you have sshed into the car, start teleop to start the laser scanner and control the car.
+```bash
+$ roslaunch mushr_base teleop.launch
+```
+Verify that the controller is working and you can move the car. If not, try reconnecting the controller and/or restarting teleop.
+
+In a separate terminal sshed into the robot, using the above process, start slam. Slam will take the data gathered from the laser 
+scanner started by teleop and transform that into a 2D map. 
+```bash
+$ roslaunch mushr_base slam.launch
```

Comment:
Put a note about making sure ROS_IP is set here. If it is not, the map won't show up in the next step